### PR TITLE
WT-8256 Create new tests capturing different verbose configuration scenarios

### PR DESCRIPTION
**Pull request by etienneptl**
Implementation of new tests to verify the new verbose API is working as expected when specific categories along with verbosity levels are set. The main changes are:

- Creation of a class that defines common helpers that are used across the different verbose test classes.
- Creation of a new file that contains the tests related to the new verbose API

### CODE REVIEW

File: test/suite/test_verbose01.py [hunk 1]
```diff
@@ -32,12 +32,9 @@
...
 
-# test_verbose01.py
-# Tests that basic uses of the verbose configuration API work as intended i.e. passing
-# single & multiple valid and invalid verbose categories. This test is mainly focused on uses
-# of the interface prior to the introduction of verbosity levels, ensuring 'legacy'-style
-# uses of the interface are still supported.
-class test_verbose01(wttest.WiredTigerTestCase, suite_subprocess):
+# Shared base class used by verbose tests.
+class test_verbose_base(wttest.WiredTigerTestCase, suite_subprocess):
+    collection_cfg = 'key_format=S,value_format=S'
```
Comment (alisonfel) [inline test/suite/test_verbose01.py:37]: I think this should live in the other class `test_verbose01` & `test_verbose02`, given its specific to the operations occurring in that test classes.

File: test/suite/test_verbose01.py [hunk 2]
```diff
@@ -63,9 +60,12 @@ def expect_verbose(self, categories, patterns):
         # to ensure we've only generated verbose messages for the expected categories.
         verbose_messages = output.splitlines()
 
-        # If no categories were passed, we aren't expecting any output.
-        if len(categories) == 0:
+        # Check if we are expecting any output. We should not expect anything if no categories were
+        # passed.
+        if len(categories) == 0 or not expect_output:
```
Comment (alisonfel) [inline test/suite/test_verbose01.py:65]: I think we can just drop `len(categories) == 0`, and just rely on `expect_output`. Should simplify the logic a bit.

File: test/suite/test_verbose01.py [hunk 3]
```diff
@@ -133,17 +139,31 @@ def test_verbose_none(self):
...
     def test_verbose_invalid(self):
         self.close_conn()
         self.assertRaisesHavingMessage(wiredtiger.WiredTigerError,
                 lambda:self.wiredtiger_open(self.home, 'verbose=[test_verbose_invalid]'),
-                '/\'test_verbose_invalid\' not a permitted choice/')
+                '/\'test_verbose_invalid\' not a permitted choice for key \'verbose\'/')
+
+    # Test use cases passing invalid verbose levels, ensuring the appropriate error message is
+    # raised.
+    def test_verbose_level_invalid(self):
```
Comment (alisonfel) [inline test/suite/test_verbose01.py:157]: I think this test should live in `test_verbose02`, as `test_verbose01` is just about the legacy API and this is testing edge-cases in the new API.

Comment (alisonfel) [reply test/suite/test_verbose01.py:157]: Oh wait, this can be removed as its already in `test_verbose02`

**Review by alisonfel — CHANGES_REQUESTED** [review]

File: test/suite/test_verbose01.py (hunk 1 above)
Comment (etienneptl) [reply test/suite/test_verbose01.py]: You're right, modified in ce5fdc7f5dc687a77f551b3c675d0bc00024e874. Please note that it was already defined in `test_verbose02`, I just had to mode the definition to `test_verbose01`.

File: test/suite/test_verbose01.py (hunk 2 above)
Comment (etienneptl) [reply test/suite/test_verbose01.py]: It will be easier to understand indeed, fixed in b98c5dce17865b46e102527fddc7b85d2d2a7673.

File: test/suite/test_verbose01.py (hunk 3 above)
Comment (etienneptl) [reply test/suite/test_verbose01.py]: Thanks, I missed that one! Removed in 357e2c72997a72c6bff6acce6471aa60b04039a8.

File: test/suite/test_verbose01.py [hunk 4]
```diff
@@ -58,13 +52,16 @@ def expect_verbose(self, categories, patterns):
...
-        output = self.readStdout(self.nlines)
+        nlines = 30000
+        output = self.readStdout(nlines)
         # Split the output into their individual messages. We want validate the contents of each message
         # to ensure we've only generated verbose messages for the expected categories.
         verbose_messages = output.splitlines()
 
-        # If no categories were passed, we aren't expecting any output.
-        if len(categories) == 0:
+        # Check if we are expecting any output.
```
Comment (ajmorton) [inline test/suite/test_verbose01.py:61]: Nitpick: This comment doesn't add much context and can be dropped

File: test/suite/test_verbose02.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+            session = conn.open_session()
+            session.create(uri, self.collection_cfg)
+            c = session.open_cursor(uri)
+            c['api'] = 'api'
+            c.close()
+            session.close()
+
+        # Test passing another single verbose category, 'compact' with different verbosity levels.
+        # Since there are verbose message with the category WT_VERB_COMPACT and the verbosity levels
+        # WT_VERBOSE_INFO (0) and WT_VERBOSE_DEBUG (1), we can test them both.
```
Comment (ajmorton) [inline test/suite/test_verbose02.py:72]: I assume that setting verbosity to level 1 should also print level 0 traces. If so are we able to check this?
Similarly when we pass in a category without a level (`['api']`) we can check this prints all verbosity levels

Comment (etienneptl) [reply test/suite/test_verbose02.py:72]: We will be able to check that the expected level is printed soon. It will be done as part of WT-8260.

Example of current traces when the `api` category is enabled:
```
[1636325735:250078][32792:0x7f26afea5700], test_checkpoint01.test_checkpoint_cursor.test_checkpoint_dne(table), file:WiredTigerHS.wt, WT_CURSOR.reset: [WT_VERB_API] CALL: WT_CURSOR:reset
```

We only see the category `WT_VERB_API` which is checked by the `expect_verbose` function. The level is not being printed yet, hence we cannot check it for now.

File: test/suite/test_verbose01.py [hunk 6]
```diff
@@ -58,13 +52,16 @@ def expect_verbose(self, categories, patterns):
         # operations on the connection (for generating the expected verbose output).
         yield conn
         # Read the contents of stdout to extract our verbose messages.
-        output = self.readStdout(self.nlines)
+        nlines = 30000
```
Comment (alisonfel) [inline test/suite/test_verbose01.py:55]: Any reasons for moving this definition as a local variable?
I'd prefer to leave this definition as a field in the `test_verbose_base` class, allowing any class that extends `test_verbose_base` to override it (based on their needs)

Comment (etienneptl) [reply test/suite/test_verbose01.py:55]: My only motivation was that I always try to reduce any variable's scope, I thought we could extend its scope only when we need it.
But I am happy to revert to anticipate that need!

File: test/suite/test_verbose01.py (hunk 4 above)
Comment (etienneptl) [reply test/suite/test_verbose01.py]: Removed in ff5e87280fc27e87475afb6a82d0c949838c0ea6

File: test/suite/test_verbose01.py (hunk 6 above)
Comment (etienneptl) [reply test/suite/test_verbose01.py]: As discussed, I reverted the changes 5d5ef9f2f27d5d1c3fb60b8d5677785558bacc29.

**Review by ajmorton — APPROVED** [review]
LGTM!

**Review by alisonfel — APPROVED** [review]
LGTM!
Thanks for iterating on my feedback!
